src/modules/hyprland/workspaces.cpp

```diff
@@ -139,23 +172,95 @@ void Workspaces::remove_workspace(std::string name) {
                                 [&](std::unique_ptr<Workspace> &x) { return x->name() == name; });
 
   if (workspace == workspaces_.end()) {
+    // happens when a workspace on another monitor is destroyed
+    return;
+  }
+
+  if ((*workspace)->is_persistent()) {
     return;
   }
 
   box_.remove(workspace->get()->button());
   workspaces_.erase(workspace);
 }
 
+void Workspaces::fill_persistent_workspaces() {
+  if (config_["persistent_workspaces"].isObject() && !all_outputs()) {
+    const Json::Value persistent_workspaces = config_["persistent_workspaces"];
+    const std::vector<std::string> keys = persistent_workspaces.getMemberNames();
+
+    for (const std::string &key : keys) {
+      const Json::Value &value = persistent_workspaces[key];
+      if (value.isInt()) {
+        // value is a number => create that many workspaces for this monitor
+        // only add if either:
+        // 1. key is "*" and this monitor is not already defined in the config
+        // 2. key is the current monitor name
+        if ((key == "*" && std::find(keys.begin(), keys.end(), bar_.output->name) == keys.end()) ||
+            key == bar_.output->name) {
+          int amount = value.asInt();
+          spdlog::debug("Creating {} persistent workspaces for monitor {}", amount,
+                        bar_.output->name);
+          for (int i = 0; i < amount; i++) {
+            persistent_workspaces_to_create_.emplace_back(
+                std::to_string(monitor_id_ * amount + i + 1));
+          }
+        }
+
+      } else if (value.isArray() && !value.empty()) {
+        // value is an array => key is a workspace name
+        // values are monitor names this workspace should be shown on
+        for (const Json::Value &monitor : value) {
+          if (monitor.isString() && monitor.asString() == bar_.output->name) {
+            persistent_workspaces_to_create_.emplace_back(key);
+            break;
+          }
+        }
+      }
+    }
+  }
+}
+
+void Workspaces::create_persistent_workspaces() {
+  for (const std::string &workspace_name : persistent_workspaces_to_create_) {
+    Json::Value new_workspace;
+    new_workspace["id"] = workspace_name.starts_with("special") ? -99 : 0;
```

Comment:
only give id as 0 if name is not int, otherwise use name as id

also workspace_name == "special" since named special workspaces have the same issue as the normal named workspaces